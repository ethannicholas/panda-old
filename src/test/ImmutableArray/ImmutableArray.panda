var a := ImmutablePrimitiveArray<MutableString>()

var b:ImmutablePrimitiveArray<Int> := [1, 2, 3, 4, 5]
b[3] := 5

b[0 .. 2] := [1, 2, 3]

b.append(10)

b.length := 3