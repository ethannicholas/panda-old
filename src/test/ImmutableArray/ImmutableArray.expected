error: ImmutableArray.panda:1:14: immutable arrays may not have mutable elements
error: ImmutableArray.panda:4:6: cannot modify immutable array
error: ImmutableArray.panda:6:9: 'panda.collections.ImmutableArray<panda.core.Int32>' not define the exclusive slice assignment ([..]:=) operator for these types
error: ImmutableArray.panda:8:1: no match found for method append(panda.core.Int8) in panda.collections.ImmutableArray<panda.core.Int32>
error: ImmutableArray.panda:10:10: cannot assign to 'b.length()'
5 errors
